core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -415,8 +415,7 @@ public void setInnerHits(Map<String, InternalSearchHits> innerHits) {
         static final String INNER_HITS = "inner_hits";
     }
 
-    @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+    public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
`CompletionSuggestion.Entry.Option#innerToXContent` uses the `toInnerXContent` for rendering the hit as part of the option, instead of rendering the hit as a seperate object.